src/service-registration/service-registration.controller.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { ServiceRegistrationController } from './service-registration.controller';
+import { ServiceRegistrationService } from './service-registration.service';
+import { HttpModule } from '@nestjs/common';
+import { MongooseModule, getModelToken } from '@nestjs/mongoose';
+import { LogsSchema } from '../schema/logs.schema';
+import { ServiceSchema } from '../schema/service.schema';
+import { dbServiceMock } from '../db-mock-data/database-service-mock';
+
+describe('ServiceRegistration Controller', () => {
+  let controller: ServiceRegistrationController;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [HttpModule],
+      controllers: [ServiceRegistrationController],
+      providers: [
+        ServiceRegistrationService,
+        {
+          provide: getModelToken('service'),
+          useValue: dbServiceMock,
+        },
+      ],
+    }).compile();
+
+    controller = module.get<ServiceRegistrationController>(
+      ServiceRegistrationController,
+    );
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+
```

Comment:
ich würde auch noch die delete methide testen. 
also einfach mal service löschen und überprüfen, dass er tatsächlich enternt wurde. 
hier dürft ich auch gerne etwas kreativ werden. was passiert z.b. wenn kein zu id passende service existiert?

Reply:
Hab mal eine Test zur delete Methode erstellt aber ich erhalte den Fehler `TypeError: this.serviceModel.findByIdAndDelete is not a function`. Hast du ne Ahnung woher das kommen kann? Wahrscheinlich hat das was damit zu tun wie die Mongo Db in den Tests gemockt wird?